Feature/Tasky/v0.1.4/UiDesignAndFunctionHandling: Addressed PR comments and refactor logic

Alex-A-Fit opened the pull request:
## Feature/Tasky/v0.1.4/UiDesignAndFunctionHandling: Addressed PR comments and refactor logic

## Bug/Feature description
Addressing PR comments led me to refactor my event implementation therefore i want to make sure everything is in a better state than prior PR

## Solution description
Addressed PR comments on TextField in core UI. Now the coreUI textfield uses TextField passing in placeholder and icon as composable parameters. Adjusted textFields properties due to switch from basicTextField to TextField. Handled all of the function calls that are not related to login data layer. Cleaned up use case functions by returning data class and changing when block to if block. Separated events into numerous sealed classes to separate events from app module to login module and internal login events.

## Code Validations 
[ ] PR is under 400 lines of code Insertion and Deletion  
[X] Files are Formatted and Optimized  
[X] Clean Architecture followed  
[X] No unneccessary comments or TODOs left in codebase  
[ ] Unit Tests Added  
[X] Unit Tests NOT added and explanation why under exemption description  
[ ] UI Tests Added  
[X] UI Tests NOT added and explanation why under exemption description  
[ ] End to End Tests Added  
[X] End to End Tests NOT added and explanation why under exemption description  
[ ] Screenshot of UI added  

## Exemption Definition
Refactoring UI Logic and event handler logic

## UI Screenshots 
![image](https://user-images.githubusercontent.com/64118964/227834406-02a352a9-dbe7-4328-8ff4-97cdd24ae173.png)

![image](https://user-images.githubusercontent.com/64118964/227834485-f30c6b3c-364a-4d8b-b274-60fb93b89acf.png)

![image](https://user-images.githubusercontent.com/64118964/227834625-b79e845c-d973-4b24-8642-7fb651407bda.png)

Alex-A-Fit committed:
Feature/Tasky/v0.1.4/UiDesignAndFunctionHandling: Addressed PR comments on TextField in core UI. Now the coreUI textfield uses TextField passing in placeholder and icon as composable parameters. Adjusted textFields properties due to switch from basicTextField to TextField. Handled all of the function calls that are not related to login data layer. Cleaned up use case functions by returning data class and changing when block to if block. Separated events into numerous sealed classes to separate events from app module to login module and internal login events.

philipplackner commented on coreui/src/main/java/com/alexafit/coreui/components/textfield/TextFieldWithIcon.kt:
```diff
@@ -1,114 +1,78 @@
-                    ),
-                    tint = iconTint
-                )
+            .fillMaxWidth()
```
Here, you're forcing this reusable composable to always span the whole width, no matter where you place it. Imagine you'd want to place 2 of them next to each other. That wouldn't work here. I recommend to not hardcode behavior with `fillMaxWidth/Height()` and instead set that on the modifier parameter passed.

philipplackner commented on onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/model/domain/mapper/ValidateEmailResult.kt:
```diff
@@ -0,0 +1,6 @@
+package com.alexafit.onboardingauthdomain.model.domain.mapper
+
+data class ValidateEmailResult(
+    val emailResult: String = "",
```
What exactly would `emailResult` mean here? From your previous PR I remember it's the trimmed email, but I don't think you'd understand that just from reading this name here. I also don't find a perfect name, but maybe something like `adjustedEmail` or so works a bit better? If you disagree, feel free to keep it

philipplackner commented on onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidateEmail.kt:
```diff
@@ -1,19 +1,25 @@
-        val isValid = android.util.Patterns.EMAIL_ADDRESS.matcher(this).matches()
-        return Pair(this, isValid)
-    }
+private fun String.isEmailValid(): ValidateEmailResult {
+    val isValid = android.util.Patterns.EMAIL_ADDRESS.matcher(this).matches()
+    return ValidateEmailResult(
+        emailResult = this,
+        validEmail = isValid
+    )
```
Here, you're using an Android-specific dependency (as you can see in the imports) which doesn't really belong into domain. Instead, I'd recommend to create an abstraction for this to keep this class locally unit testable.

Every time, you have implementation details in domain and want to get rid of that, the procedure is the same:

1. You create an abstraction (interface) in domain
2. You write an implementation of that in data (in this case using Patterns.EMAIL_ADDRESS...)
3. You pass the abstraction to the constructor of your class needing it in domain
4. Now you can decide which implementation you pass when you create that class/use case
For example:

*In domain*
```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

*In data*
```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an `EmailPatternValidator` into your use case and use there.

philipplackner commented on onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidatePassword.kt:
```diff
@@ -1,28 +1,36 @@
-            }
+class ValidatePassword {
+    operator fun invoke(password: String): ValidatePasswordResult {
+        return if (password.isBlank()) {
+            ValidatePasswordResult(
+                passwordResult = password.trim(),
+                validPassword = false
+            )
```
Also if I remember correctly the backend treats a space character as a normal special character, so passwords would be allowed to start/end with a space. In the rare case a user really chooses this, this code would accept the password but trim it which will cause the backend to save a different password than the user entered

Alex-A-Fit committed:
Feature/Tasky/v0.1.4/UiDesignAndFunctionHandling: Abstracted The functionality within UseCases in the onboardingAuthDomain layer. Now the email and password inject a validator from the data layer that then determines if the provides string matches a certain criteria.

Removed hard coded fill max width from core UI composables.

Refactored use case due to implementations from data layer

renamed data model paramters for Validating Email and Password to be more readable
